app.js

```diff
@@ -0,0 +1,178 @@
+// if I search for this documentation from browser- I can´t find this section in Documentation
+//http://www.passportjs.org/concepts/authentication/password/ (given in the lesson)
+// is it because of the old version of the website? I  think this link refers to the same instructions https://www.passportjs.org/howtos/password/
+
+require('dotenv').config();
+var bcrypt = require('bcryptjs');
+const express = require('express');
+const path = require('path');
+const session = require('express-session');
+const passport = require('passport');
+const LocalStrategy = require('passport-local').Strategy;
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const MongoDBStore = require('connect-mongodb-session')(session);
+
+//created new collection in mongo db
+var store = new MongoDBStore({
+  uri: process.env.MONGO_URI,
+  collection: 'sessions',
+});
+
+// Catch errors
+store.on('error', function (error) {
+  console.log(error);
+});
+
+const mongoDb = process.env.MONGO_URI;
+mongoose.connect(mongoDb, { useUnifiedTopology: true, useNewUrlParser: true });
+const db = mongoose.connection;
+db.on('error', console.error.bind(console, 'mongo connection error'));
+
+//SET UNIQUE USERNAME -LATER
```

Comment:
@akosasante ,  I tried to make the user choose unique name and added npm package  mongoose-unique-validator,
![image](https://github.com/NatachaKey/passport-demo/assets/103755090/afc9fa3a-de84-4d2a-b336-3384b1afa0c1)

![image](https://github.com/NatachaKey/passport-demo/assets/103755090/c856145b-7b65-48fe-802e-1c94de757b00)
 now it gives me an error when i try to sign up a user with an existing name , but how can I set a user friendly message in case username already exists, I ´ve ried diferent ways but nothing worked - I get this  (will commit it now to github)
![image](https://github.com/NatachaKey/passport-demo/assets/103755090/5be7e930-a37e-47a3-84da-1cb3b66bc9cc)